### PR TITLE
Remove deprecations from options resolver usage

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | extracted from #6738 |
| License         | MIT |

### CODE REVIEW

**Comment by pamil** [comment]
Closing in favour of #6821.

File: docs/bundles/SyliusShippingBundle/custom_calculators.rst [hunk 1]
```diff
@@ -143,7 +143,7 @@ Done, we've set the default item limit to 10. Now we have to create a form type
             ;
         }
 
-        public function setDefaultOptions(OptionsResolverInterface $resolver)
+        public function setDefaultOptions(OptionsResolver $resolver)
```
Comment (rvanlaak) [inline docs/bundles/SyliusShippingBundle/custom_calculators.rst]: Should be renamed to `configureOptions` too?

File: docs/bundles/SyliusGridBundle/custom_filter.rst [hunk 2]
```diff
@@ -60,7 +60,7 @@ And the form type:
             $builder->add('stats', 'choice', array('choices' => range($options['range'][0], $options['range'][1])));
         }
 
-        public function setDefaultOptions(OptionsResolverInterface $resolver)
+        public function setDefaultOptions(OptionsResolver $resolver)
```
Comment (rvanlaak) [inline docs/bundles/SyliusGridBundle/custom_filter.rst]: Should be renamed to `configureOptions` too?

File: docs/bundles/SyliusTaxonomyBundle/categorization.rst [hunk 3]
```diff
@@ -99,7 +99,7 @@ it is handy to use `sylius_taxon_choice` form type:
             $builder->add('taxons', 'sylius_taxon_choice');
         }
 
-        public function setDefaultOptions(OptionsResolverInterface $resolver)
+        public function setDefaultOptions(OptionsResolver $resolver)
```
Comment (rvanlaak) [inline docs/bundles/SyliusTaxonomyBundle/categorization.rst]: Should be renamed to `configureOptions` too?

File: docs/bundles/SyliusGridBundle/custom_filter.rst [hunk 4]
```diff
@@ -23,7 +23,7 @@ To add a new filter, we need to create an appropriate class and form type.
             // $data['stats'] contains the submitted value!
         }
 
-        public function setOptions(OptionsResolverInterface $resolver)
+        public function setOptions(OptionsResolver $resolver)
```
Comment (rvanlaak) [inline docs/bundles/SyliusGridBundle/custom_filter.rst]: `setOptions` might has to be renamed to `configureOptions`?

**Comment by pamil** [comment]
@rvanlaak thanks, I'll include it in #6821 🌮

**Pull request closed without merging**